### ISSUE TITLE
False positive `invalid-assignment` on every member of an enum with a specialized generic data-type mixin

### ISSUE THREAD

**Issue by gorewilliams** [astral-sh/ty#4403]
## Summary

When an enum inherits a data-type mixin that is a specialized generic alias (`Mixin[str]`), every
member is reported as incompatible with the mixin's `__new__`, regardless of the member value or the
parameter types. The same mixin written as a non-generic class is accepted.

The rejected argument is not the member value - it is the synthesized `cls` argument, so a mixin
whose `__new__` accepts `*args: object` is rejected too. Since it fires once per member, a shared
generic enum base gets very loud: one such base accounted for 80% of the diagnostics in our codebase.

This is not the "generic enum" case rejected by #2416 - these enums are not generic. The mixin is
specialized in the base-class list, so the enum has no free type parameters, and `invalid-generic-enum`
is unaffected.

## Reproducer

`ty check --python-version 3.12 repro.py`

```py
from enum import StrEnum
from typing import TYPE_CHECKING, Self


class ConcreteMixin:
    if TYPE_CHECKING:
        def __new__(cls, value: str) -> Self: ...


class GenericMixin[T]:
    if TYPE_CHECKING:
        def __new__(cls, value: T) -> Self: ...


class VariadicMixin[T]:
    if TYPE_CHECKING:
        def __new__(cls, *args: object, **kwargs: object) -> Self: ...


class Ok(ConcreteMixin, StrEnum):
    A = "a"  # no diagnostic


class Bad(GenericMixin[str], StrEnum):
    A = "a"  # error: [invalid-assignment]


class AlsoBad(VariadicMixin[str], StrEnum):
    A = "a"  # error: [invalid-assignment]
```

## Actual

```
error[invalid-assignment]: Enum member `A` is incompatible with `__new__`
info: Expected compatible arguments for `def __new__[Self](cls, value: T@GenericMixin) -> Self`

error[invalid-assignment]: Enum member `A` is incompatible with `__new__`
info: Expected compatible arguments for `def __new__[Self](cls, *args: object, **kwargs: object) -> Self`
```

## Expected

No diagnostic on either. `T` is `str`, and `"a"` is assignable to `str` and to `*args: object`. Both
classes are valid at runtime and behave identically to `Ok` (`Bad.A.value == "a"`,
`Bad("a").name == "A"`). Because `__new__` is declared under `if TYPE_CHECKING`, the mixin is not a
data type at runtime, so the "too many data types" `TypeError` does not apply. pyright reports no
errors here.

## Root cause

`inherited_user_defined_mixin_new` (`crates/ty_python_semantic/src/types/enums.rs`) walks the MRO and
calls `class.class_literal(db).as_static()` before looking up `__new__`, discarding the specialization
carried by a `GenericAlias` base. The returned `__new__` still has the mixin's typevars free, so the
implicit `cls: type[Self]` has upper bound `Mixin[T@Mixin]`, which the enum class does not satisfy -
inference fails before the member value is looked at.

## Version

ty 0.0.75 (00199f0aa 2026-08-22)

**Comment by carljm** [comment]
Confirmed, thank you!

**Comment by gorewilliams** [comment]
Thx @carljm. Opened a PR that should fix it.
